packages/core/src/odata-common/selectable/collection-field.ts

```diff
@@ -1,40 +1,57 @@
 import { Entity } from '../entity';
 import { EdmTypeShared } from '../edm-types';
-import { SelectableEdmTypeField } from './edm-type-field';
-import { Field } from './field';
-import { ComplexTypeNamespace } from './complex-type-namespace';
-import { getEntityConstructor } from './complex-type-field';
+import { Field, FieldOptions } from './field';
 import { ConstructorOrField } from './constructor-or-field';
+import { ComplexTypeField, getEntityConstructor } from './complex-type-field';
+import { ComplexTypeNamespace } from './complex-type-namespace';
 
 /**
- *
- * Represents a static field of an entity or complex type.
- *
- * @typeparam EntityT - Type of the entity the field belongs to
- * @typeparam FieldT - Type of the entries of the collection in the field
+ * Represents a field of an entity or a complex type, that can have a collection as value.
+ * @typeparam EntityT - Type of the entity the field belongs to.
+ * @typeparam CollectionFieldT - Type of of elements of the collection. This can either be an EDM type or complex type.
+ * @typeparam NullableT - Boolean type that represents whether the field is nullable.
+ * @typeparam SelectableT - Boolean type that represents whether the field is selectable.
  */
 export class CollectionField<
-    EntityT extends Entity,
-    FieldT extends EdmTypeShared<'any'> | Record<string, any> = any
-  >
-  extends Field<EntityT>
-  implements SelectableEdmTypeField
-{
-  readonly selectable: true;
-
+  EntityT extends Entity,
+  CollectionFieldT extends EdmTypeShared<'any'> | Record<string, any> = any,
+  NullableT extends boolean = false,
+  SelectableT extends boolean = false
+> extends Field<EntityT, NullableT, SelectableT> {
   /**
    *
    * Creates an instance of CollectionField.
    *
    * @param fieldName - Actual name of the field used in the OData request.
-   * @param fieldOf - The constructor of the entity or the complex type field this field belongs to.
-   * @param _fieldType - Type of the field according to the metadata description.
+   * @param _fieldOf - The constructor of the entity or the complex type field this field belongs to.
+   * @param _fieldType - Edm type of the field according to the metadata description.
+   *  @param fieldOptions - Optional settings for this field.
    */
   constructor(
     fieldName: string,
-    fieldOf: ConstructorOrField<EntityT>,
-    readonly _fieldType: FieldT | ComplexTypeNamespace<FieldT>
+    readonly _fieldOf: ConstructorOrField<EntityT>,
+    readonly _fieldType: CollectionFieldType<CollectionFieldT>,
+    fieldOptions?: Partial<FieldOptions<NullableT, SelectableT>>
```

Comment:
I see very often the `Partial<FieldOptions...` in the constructor arguments. Why not make the properties optional in the `FiledOptions` type?

Reply:
In the end I need to have all types somewhere. They cannot be optional in the type in the end, because then their type is not `true` or `false` as I need it to be, but `true | undefined` or `false | undefined`.
I am pretty sure I could achieve this by turning it around and using `Required<...>` in the end. I initially decided against that, because I think that semantically the field options are required by default and only the constructors have some sensible defaults.